dissect/target/plugins/os/windows/catroot.py

```diff
@@ -1,25 +1,46 @@
-from asn1crypto import algos, core
+from binascii import hexlify
+from collections import defaultdict, namedtuple
+from enum import Enum
+from typing import Optional
+
+from asn1crypto.cms import ContentInfo
+from asn1crypto.core import Sequence
+from dissect.esedb import EseDB
 from flow.record.fieldtypes import digest
 
 from dissect.target.exceptions import UnsupportedPluginError
 from dissect.target.helpers.record import TargetRecordDescriptor
 from dissect.target.plugin import Plugin, export
 
-HINT_NEEDLE = b"\x1e\x08\x00H\x00i\x00n\x00t"
-MD5_NEEDLE = b"\x06\x08\x2a\x86\x48\x86\xf7\x0d\x02\x05"
-SHA1_NEEDLE = b"\x06\x05\x2b\x0e\x03\x02\x1a"
-SHA_GENERIC_NEEDLE = b"\x06\x09\x08\x86\x48\x01\x65\x03\x04\x02"
+NEEDLES = {
+    "hint": b"\x1e\x08\x00H\x00i\x00n\x00t",
+    "package_name": b"\x06\n+\x06\x01\x04\x01\x827\x0c\x02\x01",
+    "md5": b"\x06\x08\x2a\x86\x48\x86\xf7\x0d\x02\x05",
+    "sha1": b"\x06\x05\x2b\x0e\x03\x02\x1a",
+    "sha_generic": b"\x06\x09\x08\x86\x48\x01\x65\x03\x04\x02",
+    "sha256": b"\x06\t`\x86H\x01e\x03\x04\x02\x01\x05\x00",
+}
+
 
 CatrootRecord = TargetRecordDescriptor(
     "windows/catroot",
     [
-        ("digest", "digest"),
-        ("path", "hint"),
+        ("digest[]", "digests"),
```

Comment:
Maybe nicer to yield individual records per digest? Makes the post-processing (what this plugin will likely end up being used for) a lot easier and nicer.

Reply:
And in case there are multiple hints, would you like me to add all the hints to each (digest) record? Since I don't know at the moment how the hints are related to the digests

Reply:
Ideally only the related hints, but if you don't know yet, then yes.

Reply:
Done! Added a comment as well